lms/djangoapps/bulk_email/tasks.py

```diff
@@ -220,16 +171,23 @@ def perform_delegate_email_batches(entry_id, course_id, task_input, action_name)
     course = get_course(course_id)
 
     # Get arguments that will be passed to every subtask.
-    to_option = email_obj.to_option
+    targets = email_obj.targets.all()
     global_email_context = _get_course_email_context(course)
 
-    recipient_qsets = _get_recipient_querysets(user_id, to_option, course_id)
+    recipient_qsets = [
+        target.get_users(course_id, user_id)
+        for target in targets
+    ]
+    combined_set = User.objects.none()
+    for qset in recipient_qsets:
+        combined_set |= qset
+    combined_set = combined_set.distinct()
```

Comment:
I thought you were right about this, but it turns out the `distinct` is still needed to guarantee no duplicate emails are sent.

Failing test, notice that `robot+test+212@edx.org` is recipient 14 and 15
https://build.testeng.edx.org/job/edx-platform-python-unittests-pr/19262/testReport/lms.djangoapps.bulk_email.tests.test_email/TestEmailSendFromDashboardMockedHtmlToText/test_no_duplicate_emails_staff_instructor/